### PR TITLE
chore: contract interfaces migration

### PR DESCRIPTION
**Pull request by Jonathansumner**
Migration regarding #83

### CODE REVIEW

File: migrations/committed/000002.sql [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+      }
+      plv8.execute("insert into contracts (id, interfaces, store_contract_message_id, instantiate_message_id) ($1, {$2}, $3, $4)", [contract_address, getJaccard(JSON.parse(payload_stripped)), id, store_code_id]);
+    }
+  }
+  return __toCommonJS(current_exports);
+})();
+
+return plv8ify.migrationAddInterfaceSupport()
+
+$plv8ify$ LANGUAGE plv8 IMMUTABLE STRICT;
```
Comment (bryanchriswhite) [inline migrations/committed/000002.sql:188]: This will define the migration function but it still needs to be called:

```sql
SELECT * from plv8ify_migrationAddInterfaceSupport();
```

File: migrations/initial_schema.sql [hunk 2]
```diff
@@ -3,7 +3,7 @@
 --
 
 -- Dumped from database version 14.5 (Debian 14.5-2.pgdg110+2)
--- Dumped by pg_dump version 14.5
+-- Dumped by pg_dump version 14.5 (Ubuntu 14.5-0ubuntu0.22.04.1)
```
Comment (bryanchriswhite) [inline migrations/initial_schema.sql:6]: Ignore what I wrote in the README about updating this file. It only needs to represent the state of the DB pre-migrations and will not be updated. 

When I get around to it, we will start tracking the "current" schema in a file, which we will produce at the end of migrations such that anyone can restore the current schema directly (as opposed to applying the initial SQL and then all migrations).

TLDR; let's revert changes to this file for now.

Comment (Jonathansumner) [reply migrations/initial_schema.sql:6]: Ahh sorry I think I must have misunderstood - I get you now

Comment (bryanchriswhite) [reply migrations/initial_schema.sql:6]: Nope, pretty sure it was just me doing a full 180 in the docs. :sweat_smile:

File: migrations/committed/000002.sql [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+
+SELECT * from migrationAddInterfaceSupport();
```
Comment (bryanchriswhite) [inline migrations/committed/000002.sql:190]: The name of the function as it's stored in Postgres ([000002.sql:8](https://github.com/fetchai/ledger-subquery/pull/174/files#diff-be76b025d211974f65a936059905d14123deddad3953db07871094d9c34a4540R8)) has a prefix (which I think we would have to remove manually, not sure it's worth it):
> ```sql
> CREATE OR REPLACE FUNCTION plv8ify_migrationAddInterfaceSupport() RETURNS JSONB AS $plv8ify$
> ```

```suggestion
SELECT * from plv8ify_migrationAddInterfaceSupport();
```

Comment (bryanchriswhite) [reply migrations/committed/000002.sql:190]: I think it may add clarity to make a slight modification to the name; for example, "add contract interface detection" as opposed to "add interface support".

File: migrations/current.ts [hunk 4]
```diff
@@ -1,5 +1,186 @@
-// write plv8 based migration here
```
Comment (bryanchriswhite) [inline migrations/current.ts:1]: My apologies for the in-progress nature of the README on this stuff. :sweat_smile: 

This file is where you would work on your migration locally but once you `graphile-migrate commit`, we should copy it to a file in the migrations/src directory with a name corresponding to it's respective .sql file in migrations/committed and then current.ts.

**Comment by bryanchriswhite** [comment]
As discussed, this migration isn't necessary given the starting point for migrations in production. Thanks nonetheless @Jonathansumner!
